pkg/internal/testhelpers/containers.go

```diff
@@ -71,37 +72,28 @@ func (e *ExtensionState) UseMultinode() {
 	*e |= timescaleBit | timescale2Bit | multinodeBit
 }
 
-func (e ExtensionState) usesTimescaleDB() bool {
-	switch e {
-	case VanillaPostgres:
-		return false
-	default:
-		return true
-	}
+func (e *ExtensionState) UsePG12() {
+	*e |= postgres12Bit
+}
+
+func (e ExtensionState) UsesTimescaleDB() bool {
+	return (e & timescaleBit) != 0
+}
+
+func (e ExtensionState) UsesTimescale2() bool {
+	return (e & timescale2Bit) != 0
 }
 
 func (e ExtensionState) UsesMultinode() bool {
-	switch e {
-	case Multinode:
-		return true
-	case MultinodeAndPromscale:
-		return true
-	default:
-		return false
-	}
+	return (e & multinodeBit) != 0
 }
 
 func (e ExtensionState) usesPromscale() bool {
-	switch e {
-	case Timescale1AndPromscale:
-		return true
-	case Timescale2AndPromscale:
-		return true
-	case MultinodeAndPromscale:
-		return true
-	default:
-		return false
-	}
+	return (e & promscaleBit) != 0
+}
+
+func (e ExtensionState) UsesPG12() bool {
```

Comment:
[nit] can we have this near to `UsePG12()`?

Reply:
That's not the pattern in the file, and redoiing everything would be a larger change than I want right now